src/Avalonia.Ide.CompletionEngine/CompletionEngine.cs

```diff
@@ -128,7 +128,14 @@ public MetadataProperty LookupProperty(string typeName, string propName)
                 var xmlRdr = XmlReader.Create(new StringReader(xml));
                 while (xmlRdr.NodeType != XmlNodeType.Element)
                 {
-                    xmlRdr.Read();
+                    try
+                    {
+                        xmlRdr.Read();
```

Comment:
It's probably better to also check for the return value here. Also, I think exception doesn't need to be rethrown.

Reply:
example sutiation:
```xaml
<UserControl xmlns="https://github.com/avaloniaui" 
              xmlns:my="mynamespace-valid-url-or-clr"
               xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
my: (user typing completion works fine in this pr...)
>
```
state of the xaml parser:

- `xmlRdr.Read() => false` and exception thrown 
- `xmlRdr.NodeType => Element`
- `xmlRdr.AttributeCount => 3` 3 valid attributes with namespaces are parsed

so the idea is we don't loose information about 3 valid attributes parsed and processed fine with xmlparser.
even Parse resulted `false`, 3 attribute values are there and good.
So i don't think there is need to check result of Read as it's meaningful for the hole element,  but what we need here is the namespaces declaration/s mostly.

@kekekeks let me know if you still think we should check Read() result and why, i can skip rethrow with a flag no problem, we'll save few ns ;)